--- a/share/check_mk/modules/nagios.py	2018-02-15 22:16:16.905602616 +0100
+++ b/share/check_mk/modules/nagios.py	2018-02-15 22:16:36.689621413 +0100
@@ -1272,13 +1272,10 @@
             console.verbose(" (new content)", stream=sys.stderr)
 
     os.rename(source_filename + ".new", source_filename)
-    if not delay_precompile:
-        py_compile.compile(source_filename, compiled_filename, compiled_filename, True)
-        os.chmod(compiled_filename, 0755)
-    else:
-        if os.path.exists(compiled_filename) or os.path.islink(compiled_filename):
-            os.remove(compiled_filename)
-        os.symlink(hostname + ".py", compiled_filename)
+
+    if os.path.exists(compiled_filename) or os.path.islink(compiled_filename):
+        os.remove(compiled_filename)
+    os.symlink(hostname + ".py", compiled_filename)
 
     console.verbose(" ==> %s.\n", compiled_filename, stream=sys.stderr)
 
